src/main/java/org/takes/tk/TkProxy.java

```diff
@@ -0,0 +1,144 @@
+/**
+ * The MIT License (MIT)
+ *
+ * Copyright (c) 2015 Yegor Bugayenko
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included
+ * in all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+package org.takes.tk;
+
+import java.io.IOException;
+import lombok.EqualsAndHashCode;
+import lombok.ToString;
+import org.takes.Request;
+import org.takes.Response;
+import org.takes.Take;
+
+/**
+ * Proxy take.
+ *
+ * <p>This take may transform the request before passing it
+ * to the original take and/or transform the response received
+ * from the original take.
+ *
+ * <p>If the request transformer is not provided, the original
+ * request is passed to the original take.
+ *
+ * <p>If the response transformer is not provided, the response
+ * received from the original take is returned.
+ *
+ * <p>The class is immutable and thread-safe.
+ *
+ * @author Dragan Bozanovic (bozanovicdr@gmail.com)
+ * @version $Id$
+ * @since 0.25
+ */
+@ToString
+@EqualsAndHashCode
```

Comment:
@bdragan By default, @EqualsAndHashCode use all non-static, non-transient fields. So `#equals()` and `#hashCode()` will be invalid here.  You should mention transient field names in the annotation.